### PR TITLE
More Github actions

### PR DESCRIPTION
**Pull request by grugnog**
This makes a few improvements:

- Adds preview links to PRs again (using the RTD integration instead of the old method)
- Adds a checker for external links
- Fixes all the broken external links (whew!)
- Simplify the remark lint configs by using some presets as a base
- Fix the resulting remark lint errors (whew!)
- Add a weekly check of master branch for broken links and remark regressions
- Renames workflows to more helpful names

----
:books: Documentation preview :books:: https://civicactions-handbook--996.org.readthedocs.build/en/996/

### CODE REVIEW

File: .github/workflows/preview-links.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: Read the Docs Pull Request Preview
+on:
+  pull_request_target:
+    types:
+      - opened
```
Comment (dmundra) [inline .github/workflows/preview-links.yml]: @grugnog I didn't see this workflow run. Not sure why it didn't.

Comment (grugnog) [reply .github/workflows/preview-links.yml]: This is fixed now

**Comment by grugnog** [comment]
@dmundra I think this is ready to review now - the remark/reviewdog check is still failing, but that is only due to their being so many suggestions (due to a lot of files being touched in the PR) it hits a Github API limit, so isn't going to affect normal PRs. There is a fix for this upstream in Reviewdog but it's not released yet, so I suggest for now we temporarily change the branch protection rules to permit this, then re-enable once merged.
Like the other PRs that are really broad and there aren't any meaningful content changes, we will need to use admin powers to force merge.
I have tested the weekly check via temporary commits (now rebased out) and [all the steps](https://github.com/CivicActions/guidebook/actions/workflows/check-all.yml) are working as expected, so that should just report to Slack if/when anything breaks on master.

**Review by dmundra — APPROVED** [review]

**Comment by dmundra** [comment]
@grugnog sounds good. Approved!

**Pull request merged**